### ISSUE TITLE
Whitespaces are incorrect stripped from text

### ISSUE THREAD

**Issue by upsuper** [magiclen/html-minifier#1]
Test case:
```rust
html_minifier::minify("<a>1</a>\n /\n <a>2</a>")
```
in this example, the result should be `<a>1</a>␣/␣<a>2</a>` however it outputs `<a>1</a>/<a>2</a>`, i.e. whitespaces around `/` are incorrectly stripped.

**Comment by magiclen** [comment]
Hi, I wonder that

```html
<p>
     This is the first line.
     This is the second line.
     This is the third line.
</p>
```
would be correct for you if its minified html were
```html
<p>This is the first line. This is the second line. This is the third line.</p>
```

, right?

**Comment by upsuper** [comment]
That's right, but it's correct because `<p>` is a block level element and thus spaces on its boundary are useless, however `<a>` is generally considered an inline level element, so spaces around it do matter.

**Comment by magiclen** [comment]
Okay, I know.

In Chinese, we prefer not to separate each word with spaces. For example,

```html
<p>
     這是第一行。
     這是第二行。
     這是第三行。
</p>
```

If it was minified to

```html
<p>這是第一行。 這是第二行。 這是第三行。</p>
```

, that would not be so correct.

For the same reason,

```html
<body>
    <a>這是第一行。</a>
    這是第二行。
    <a>這是第三行。</a>
</body>
```

I think 

```html
<body><a>這是第一行。</a>這是第二行。<a>這是第三行。</a></body>
```

is more correct.


However, in your example,

```html
<a>1</a>
 / 
<a>2</a>
```

There are spaces beside the `/` character. Considering its parent element and the indention, it looks like

```html
<body>
    <a>1</a>
     / 
    <a>2</a>
</body>
```

If we want to use this crate to minify it to

```html
<body><a>1</a> / <a>2</a></body>
```

this crate needs to be implemented and provide at least one of these options below

1. Check the indention whether it is using an excessive number of tabs or spaces and remain the surplus.

2. Combine consecutive newline characters and space characters to one space character.

**Comment by upsuper** [comment]
If you want to take collapsing whitespaces between CJ characters in to account, you may want to read the [related section](https://drafts.csswg.org/css-text-3/#white-space-processing) in CSS Text spec. This section is still under discussion, though.

The general idea there is, if there is a line break, and the last non-ws character before the break or the first non-ws character after the break is a Chinese / Japanese character, then the line break as well as all whitespaces around it are stripped.

This doesn't apply to latin characters anyway.

**Comment by magiclen** [comment]
I've rewritten half of the code to try not to let minification affect the look of the original HTML page. It should be more correct now.

**Comment by magiclen** [comment]
This issue has been fixed since `2.0.0`.
